tests/integration_tests/tests/agent_tests/test_plugin_update.py

```diff
@@ -17,63 +17,258 @@
+        self._upload_v_2_plugin()
 
-    def _upload_v_2_0_plugin(self):
-        source_dir = resource('plugins/{0}'.format(self.plugin_name))
-        target_dir = os.path.join(self.workdir, self.plugin_name)
+        # Execute base (V 1.0) host op
+        for dep_id in self.setup_deployment_ids:
+            self.setup_deployment_id = dep_id
+            self._execute_workflows()
+            self._assert_host_values(self.versions[0])
+
+        self._perform_plugins_update()
+
+        # Execute mod (V 2.0) host op
+        for dep_id in self.setup_deployment_ids:
+            self.setup_deployment_id = dep_id
+            self._execute_workflows()
+            self._assert_host_values(self.versions[1])
+
+    def _perform_plugins_update(self):
+        execution_id = self.client.plugins_update.update_plugins(
+            self.base_blueprint_id).execution_id
+        execution = self.client.executions.get(execution_id)
+        self.wait_for_execution_to_end(execution)
+
+    def _prepare_files(self):
+        # Copy v1.0 twice to different directories
+        source_dir = self._get_source_dir_for_plugin()
+        self._copy_plugin_files_to_resources(source_dir, self.versions[0])
+        target_dir = self._copy_plugin_files_to_resources(source_dir,
+                                                          self.versions[1])
+        # Replace 1.0 strings with 2.0
+        self._replace_version(target_dir, *self.versions)
+
+    def _copy_plugin_files_to_resources(self, source_dir, version):
+        target_dir = os.path.join(self.dsl_resources_path,
+                                  'plugins',
+                                  self.plugin_dir_name_prefix + version)
         shutil.copytree(source_dir, target_dir)
+        return target_dir
 
-        self._replace_version(target_dir)
-        self.upload_mock_plugin(self.plugin_name, plugin_path=target_dir)
+    def _remove_files(self):
+        for version in self.versions:
+            dir_to_rm = os.path.join(
+                self.dsl_resources_path,
+                'plugins',
+                self.plugin_dir_name_prefix + version)
+            shutil.rmtree(dir_to_rm, ignore_errors=True)
+
+    def _perform_update(self, update_plugins=True):
+        execution_id = \
+            self.client.deployment_updates.update_with_existing_blueprint(
+                deployment_id=self.setup_deployment_id,
+                blueprint_id=self.mod_blueprint_id,
+                update_plugins=update_plugins
+            ).execution_id
+        execution = self.client.executions.get(execution_id)
+        self.wait_for_execution_to_end(execution)
+
+    def _get_work_dir_for_plugin(self):
+        return os.path.join(self.workdir, self.plugin_name)
+
+    def _get_source_dir_for_plugin(self):
+        plugin_dir = '{0}'.format(self.plugin_name)
+        return resource(os.path.join('plugins', plugin_dir))
+
+    def _copy_plugin_files_to_work_dir(self):
```

Comment:
is there a test that the temp blueprint is really deleted?

Reply:
[Yep](https://github.com/cloudify-cosmo/cloudify-manager/blob/77dfdd9179c6be859320fe461aa6874757c3b94e/rest-service/manager_rest/test/endpoints/test_plugins_updates.py#L242)